crates/rpc/tests/rpc.rs

```diff
@@ -0,0 +1,477 @@
+// These tests need Beerus to run in the background, hence why they're hidden behind the following feature.
+#![cfg(feature = "integration-tests")]
+
+use cached::Cached;
+use std::ops::Deref;
+
+use cached::SizedCache;
+use reqwest::Url;
+use starknet::{
+    core::types::{
+        BlockId, BlockWithTxHashes, BlockWithTxs, DeclareTransaction,
+        DeployAccountTransaction, FieldElement, InvokeTransaction,
+        MaybePendingBlockWithTxHashes, MaybePendingBlockWithTxs,
+        MaybePendingTransactionReceipt, Transaction, TransactionReceipt,
+    },
+    providers::{
+        jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError,
+    },
+};
+
+struct TestContext<T> {
+    client: TestClient,
+    block: BlockWithTxs,
+    block_id: BlockId,
+    extracted_value: T,
+}
+
+/// Creates a `TestContext` with the latest block.
+async fn latest_block_context() -> TestContext<()> {
+    context(|_block| Some(())).await
+}
+
+/// Creates a `TestContext` with the latest block with at least N transactions.
+async fn n_txs_context(min_tx_number: usize) -> TestContext<()> {
+    context(|block| {
```

Comment:
Just as an alternative way of caching: here might be the least intrusive point to cache the result if necessary (maybe with [OnceLock](https://doc.rust-lang.org/std/sync/struct.OnceLock.html)).

Reply:
This does look more complicated to me ^^'